### DISCUSSION TITLE
Email Notifications can't get to work

### DISCUSSION THREAD

**Discussion by seh2000** [alexjustesen/speedtest-tracker#1243] [Q&A] [answered]
I read this post "Unable to issue email notifications #697" which I followed for my settings, but I am getting the error:
<admin@Pi-hole1:~/docker/speedtest-tracker $ sudo docker-compose up -d
ERROR: Invalid interpolation format for "environment" option in service "speedtest-tracker": "MAIL_PASSWORD="20Xxxxxxxxx$xx"">
Read on the internet it could related to if one had the dollar sign in the password then one needed to add a second dollar sign.
I did add the extra $ to the password MAIL_PASSWORD="20Xxxxxxxxx$**$**xx" and I got no error but no email is sent/received.

Below my configuring - what have I made a mistake?
         environment:
            - PUID=1000
            - PGID=1000
            - DB_CONNECTION=mysql
            - DB_HOST=db
            - DB_PORT=3306
            - DB_DATABASE=speedtest_tracker
            - DB_USERNAME=speedy
            - DB_PASSWORD=password
#        mail:
            - MAIL_DRIVER=smtp
            - MAIL_MAILER=smtp
           - MAIL_HOST=smtp.gmail.com
            - MAIL_PORT=587
            - MAIL_USERNAME="mygmail@gmail.com"
            - MAIL_PASSWORD="20Xxxxxxxxx$xx"
            - MAIL_ENCRYPTION=TLS
            - MAIL_FROM_ADDRESS="mygmail@gmail.com"
           - MAIL_FROM_NAME="Speedtest Tracker"
        volumes:
#           - speedtest-app:/config
            - /home/admin/docker/speedtest-tracker
            - '/path/to/directory/web:/etc/ssl/web'
        image: lscr.io/linuxserver/speedtest-tracker:0.17.2

**Comment by alexjustesen** [comment]
Drop `- '/path/to/directory/web:/etc/ssl/web'` from your docker-compose.yml file, this isn't a valid path and you only need it if you're passing in custom certificates.

**Reply by seh2000** [reply]
I dropped the path for custom certificates and added the 2nd $-sign to the password as with out I get this "interpolation" format error, "docker-compose up -d" runs without errors, but no email is sent.  


version: '3.3'
services:
    speedtest-tracker:
        container_name: speedtest-tracker
        ports:
            - '8981:80'
            - '8943:443'
        environment:
            - PUID=1000
            - PGID=1000
            - DB_CONNECTION=mysql
            - DB_HOST=db
            - DB_PORT=3306
            - DB_DATABASE=speedtest_tracker
            - DB_USERNAME=speedy
            - DB_PASSWORD=password
            - MAIL_DRIVER=smtp
            - MAIL_MAILER=smtp
            - MAIL_HOST=smtp.gmail.com
            - MAIL_PORT=587
            - MAIL_USERNAME="seh2000@gmail.com"
            - MAIL_PASSWORD="20Xxxxxxxxx$**$**16"
            - MAIL_ENCRYPTION=TLS
            - MAIL_FROM_ADDRESS="seh2000@gmail.com"
            - MAIL_FROM_NAME="Speedtest Tracker"
        volumes:
            - /home/admin/docker/speedtest-tracker
        image: lscr.io/linuxserver/speedtest-tracker:0.17.2
        restart: unless-stopped
        depends_on:
            - db
    db:
        image: mariadb:10
        restart: always
        environment:
            - MARIADB_DATABASE=speedtest_tracker
            - MARIADB_USER=speedy
            - MARIADB_PASSWORD=password
            - MARIADB_RANDOM_ROOT_PASSWORD=true
        volumes:
            - speedtest-db:/var/lib/mysql
volumes:
  speedtest-app:
  speedtest-db:

**Reply by alexjustesen** [reply]
I'll check this tonight, have you tried using port `465`?

**Reply by seh2000** [reply]
Hi Alex,
Will try.
The official settings are:
![image](https://github.com/alexjustesen/speedtest-tracker/assets/74658646/516ad980-4293-4c6d-a51c-184042acd08d)

**Reply by alexjustesen** [reply]
I think tls never worked for me and I stuck with SSL

**Reply by seh2000** [reply]
Hi Have tried now with 465, and SSL even put ssl still nothing.
Properly me who is doing something wrong.


            - MAIL_DRIVER=smtp
            - MAIL_MAILER=smtp
            - MAIL_HOST=smtp.gmail.com
            - MAIL_PORT=465
            - MAIL_USERNAME="seh2000@gmail.com"
            - MAIL_PASSWORD="20Xxxx$$16"
            - MAIL_ENCRYPTION=ssl
            - MAIL_FROM_ADDRESS="seh2000@gmail.com"
            - MAIL_FROM_NAME="Speedtest Tracker"

**Reply by alexjustesen** [reply]
Last thought, you generated an application password and you're not using your account password right?

**Reply by seh2000** [reply]
Hi Alex,
I have for other apps followed this:
_If you connect using SSL or TLS, you can send email to anyone inside or outside of your organization using smtp.gmail.com as your SMTP server. This option requires you to authenticate with your Gmail or Google Workspace account and password when you set it up. The device uses those credentials every time it attempts to send email._
However, I have just created a App password and still same.

**Reply by alexjustesen** [reply]
The following worked for me, this is with creating the app password from your Google Account Security page.

```env
MAIL_MAILER=smtp
MAIL_HOST=smtp.gmail.com
MAIL_PORT=465
MAIL_USERNAME=******@gmail.com
MAIL_PASSWORD="app password goes here"
MAIL_ENCRYPTION=tls
MAIL_FROM_ADDRESS=******@gmail.com
MAIL_FROM_NAME="Speedtest Tracker"
```

Also make sure in your `.env` file you uncomment the `MAIL_` prefixed env variables.

**Reply by seh2000** [reply]
I assume .env is my .yml file and if then yes, if not the .yml file please tell me where to find.


            - MAIL_DRIVER < I have tried with and without this one = same result 
            - MAIL_MAILER=smtp
            - MAIL_HOST=smtp-relay.gmail.com < I have tried with this and with smtp.gmail.com no difference
            - MAIL_PORT=465
            - MAIL_USERNAME="seh2000@gmail.com"
            - MAIL_PASSWORD="xxxxxxxxxx" <App Password generated by google 
            - MAIL_ENCRYPTION=SSL
            - MAIL_FROM_ADDRESS="seh2000@gmail.com"
            - MAIL_FROM_NAME="Speedtest Tracker"

I used this command "docker-compose up -d && docker-compose logs -f" where in the log I found this:

speedtest-tracker    | [2024-02-27 10:26:14] production.ERROR: Connection could not be established with host "ssl://mailhog:465": stream_socket_client(): php_network_getaddresses: getaddrinfo for mailhog failed: Name does not resolve {"exception":"[object] (Symfony\\Component\\Mailer\\Exception\\TransportException(code: 0): Connection could not be established with host \"ssl://mailhog:465\": stream_socket_client(): php_network_getaddresses: getaddrinfo for mailhog failed: Name does not resolve at /app/www/vendor/symfony/mailer/Transport/Smtp/Stream/SocketStream.php:154)
speedtest-tracker    | [stacktrace]
speedtest-tracker    | #0 [internal function]: Symfony\\Component\\Mailer\\Transport\\Smtp\\Stream\\SocketStream->Symfony\\Component\\Mailer\\Transport\\Smtp\\Stream\\{closure}()
speedtest-tracker    | #1 /app/www/vendor/symfony/mailer/Transport/Smtp/Stream/SocketStream.php(157): stream_socket_client()
speedtest-tracker    | #2 /app/www/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(275): Symfony\\Component\\Mailer\\Transport\\Smtp\\Stream\\SocketStream->initialize()
speedtest-tracker    | #3 /app/www/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(213): Symfony\\Component\\Mailer\\Transport\\Smtp\\SmtpTransport->start()
speedtest-tracker    | #4 /app/www/vendor/symfony/mailer/Transport/AbstractTransport.php(69): Symfony\\Component\\Mailer\\Transport\\Smtp\\SmtpTransport->doSend()
speedtest-tracker    | #5 /app/www/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(137): Symfony\\Component\\Mailer\\Transport\\AbstractTransport->send()
speedtest-tracker    | #6 /app/www/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(573): Symfony\\Component\\Mailer\\Transport\\Smtp\\SmtpTransport->send()
speedtest-tracker    | #7 /app/www/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(335): Illuminate\\Mail\\Mailer->sendSymfonyMessage()
speedtest-tracker    | #8 /app/www/vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(205): Illuminate\\Mail\\Mailer->send()
speedtest-tracker    | #9 /app/www/vendor/laravel/framework/src/Illuminate/Support/Traits/Localizable.php(19): Illuminate\\Mail\\Mailable->Illuminate\\Mail\\{closure}()
speedtest-tracker    | #10 /app/www/vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(198): Illuminate\\Mail\\Mailable->withLocale()
speedtest-tracker    | #11 /app/www/vendor/laravel/framework/src/Illuminate/Mail/SendQueuedMailable.php(83): Illuminate\\Mail\\Mailable->send()

**Reply by alexjustesen** [reply]
Your `.env` file will be located in the `/config` directory you mounted. After you edit it just make sure to restart the container so it picks up on the changes.

**Reply by seh2000** [reply, marked as answer]
Somehow I got it to work 
![Screenshot 2024-02-27 124048](https://github.com/alexjustesen/speedtest-tracker/assets/74658646/10d6ace0-f140-4984-9d7b-5afd282b0af2)
I used another smtp email server.

            - MAIL_MAILER=smtp
            - MAIL_HOST=smtp.dreamhost.com
            - MAIL_PORT=587
            - MAIL_USERNAME="my@email.com"
            - MAIL_PASSWORD="%20Xxxxxx$$20"
            - MAIL_ENCRYPTION=tls
            - MAIL_FROM_ADDRESS="my@email.com"
            - MAIL_FROM_NAME="Speedtest Tracker"

Must somehow be related to gmail, just not sure what...perhaps others here got some ideas/suggestions?

In the log file I found some issues, which I'll ask in other time.

**Comment by seh2000** [comment]
Just a note
I have an Outlook account, used that one = worked immediately.
Assume its something related to Gmail.
